### PR TITLE
Added UPI component fix and updated web component version to v5.28.0

### PR DESCRIPTION
**Pull request by zenit2001**
This PR includes the fix for UPI component and updates web component version to v5.28.0.

### CODE REVIEW

**Review by amihajlovski — APPROVED** [review]

File: src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/checkoutConfiguration.js [hunk 1]
```diff
@@ -267,9 +267,15 @@ function getGiftCardConfig() {
 
 function handleOnChange(state) {
   let { type } = state.data.paymentMethod;
+  const multipleTxVariantsMethods = constants.MULTIPLE_TX_VARIANT_METHODS;
+  // Bug in the UPI component, as soon as its fixed this line of code is not needed anymore
+  store.componentsObj.upi.isValid = false;
```
Comment (zaiddreakh) [inline src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/checkoutConfiguration.js:272]:
```suggestion
  store.componentsObj.upi?.isValid = false;
```

Comment (zaiddreakh) [reply src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/checkoutConfiguration.js:272]: if upi is not in the payment methods response i think it will throw an error

File: src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/checkoutConfiguration.js [hunk 2]
```diff
@@ -267,9 +267,15 @@ function getGiftCardConfig() {
...
   if (store.selectedMethod === 'googlepay' && type === 'paywithgoogle') {
     type = 'googlepay';
   }
+  if (multipleTxVariantsMethods.includes(store.selectedMethod)) {
```
Comment (zaiddreakh) [inline src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/checkoutConfiguration.js:276]: we can possibly refactor the google pay if statement into this multipleTxVariantsMethods. But can also be done at a different PR

File: src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/renderPaymentMethod.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 const store = require('../../../../store');
 const helpers = require('./helpers');
 const constants = require('../constants');
+const { MULTIPLE_TX_VARIANT_METHODS } = require('../constants');
```
Comment (maassenbas) [inline src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/renderPaymentMethod.js:4]: Instead of this line, we can just use `constants. MULTIPLE_TX_VARIANT_METHODS`

**Review by zaiddreakh — APPROVED** [review]

**Review by wboereboom — DISMISSED** [review]

File: src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/checkoutConfiguration.js [hunk 4]
```diff
@@ -267,9 +267,15 @@ function getGiftCardConfig() {
 
 function handleOnChange(state) {
   let { type } = state.data.paymentMethod;
+  const multipleTxVariantsMethods = constants.MULTIPLE_TX_VARIANT_METHODS;
   if (store.selectedMethod === 'googlepay' && type === 'paywithgoogle') {
     type = 'googlepay';
   }
+
+  if (multipleTxVariantsMethods.includes(store.selectedMethod)) {
```
Comment (maassenbas) [inline src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/checkoutConfiguration.js:275]: Can't we just use the `store.selectedMethod` as key for the `store.componentsObj` for all payment methods? This way we don't have to maintain an array with "multipleTxVariantsMethods" and it will be compatible out of the box for future components with multiple tx variants.

File: src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/renderPaymentMethod.js [hunk 5]
```diff
@@ -114,7 +114,9 @@ function hasNoChildNodes({ paymentMethodID, container }) {
 function setValid({ paymentMethodID, container }) {
   if (
     hasNoChildNodes({ paymentMethodID, container }) &&
-    ['bcmc', 'scheme'].indexOf(paymentMethodID) === -1
+    ['bcmc', 'scheme']
+      .concat(constants.MULTIPLE_TX_VARIANT_METHODS)
```
Comment (maassenbas) [inline src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/renderPaymentMethod.js:118]: Then we also don't have to change this part

**Review by maassenbas — CHANGES_REQUESTED** [review]

**Review by amihajlovski — APPROVED** [review]

**Review by wboereboom — APPROVED** [review]

**Review by maassenbas — DISMISSED** [review]

**Review by zaiddreakh — DISMISSED** [review]

**Review by zaiddreakh — APPROVED** [review]

**Review by maassenbas — APPROVED** [review]
